### PR TITLE
Get TestSet kwargs working

### LINKED ISSUE #90: TestReports does not seem to support `verbose=true` option for `@testset`
I don't know how to provide an example for this specific problem, but `TestReports.test()` fails if I use `@testset verbose = true "Something" begin
`
Any ideas why ?

### PR DESCRIPTION
**Pull request by mmiller-max**
First commit fixes an issue where all errors were being hidden when `finish` was being applied to display the results.
Second commit enables `showtime` and `verbose` kwargs to be used in package tests and the displayed output (in `display_reporting_testset`) updated to ensure correct test timings are displayed and that the kwargs have the intended effect.

Closes #90 .

### CODE REVIEW

File: src/testsets.jl [hunk 1]
```diff
@@ -339,9 +339,12 @@ function display_reporting_testset(ts::ReportingTestSet)
     try
         # Finish each of the results of the top level testset, to mimick the
         # output from Pkg.test()
-        finish.(ts_default.results)
-    catch TestSetException
+        for r in ts_default.results
+            r isa AbstractTestSet && finish(r)
+        end
+    catch err
```
Comment (oxinabox) [inline src/testsets.jl:358]: Should the try-catch be inside the `for` loop?

**Review by oxinabox — APPROVED** [review]
some minor points

File: src/testsets.jl [hunk 2]
```diff
@@ -163,6 +165,12 @@ Sets the time taken field of a `ReportingTestSet`. This is used when flattening
 """
 set_time_taken!(ts::ReportingTestSet, time_taken::Millisecond) = ts.time_taken = time_taken
 set_time_taken!(ts::AbstractTestSet, time_taken::Millisecond) = nothing
+@V1_8 function set_time_taken!(ts::DefaultTestSet, time_taken::Millisecond)
```
[the remark this answers is not in the document]
Comment (mmiller-max) [reply src/testsets.jl:168]: You're right this is clearer 👍

File: src/testsets.jl (hunk 1 above)
Comment (mmiller-max) [reply src/testsets.jl:358]: Although that would display more results, it's not how `Pkg.test` would display the results (that is if first testset of a package fails then the second one wouldn't be displayed as an error is thrown when the first testset `finish`es. So will keep it as is for now

**Review by oxinabox — APPROVED** [review]
LGTM

**Pull request merged**